internal/uvm/vsmb.go

```diff
@@ -35,6 +38,30 @@ func (vsmb *VSMBShare) Release(ctx context.Context) error {
 	return nil
 }
 
+// To clone VSMB share we just need to add it into the config doc of that VM and increase the
+// vsmb counter.
+func (vsmb *VSMBShare) Clone(ctx context.Context, vm *UtilityVM, cd *CloneData) (interface{}, error) {
+	cd.doc.VirtualMachine.Devices.VirtualSmb.Shares = append(cd.doc.VirtualMachine.Devices.VirtualSmb.Shares, hcsschema.VirtualSmbShare{
+		Name:    vsmb.Name,
+		Path:    vsmb.HostPath,
+		Options: &vsmb.Options,
+	})
+	vm.vsmbCounter++
+
+	clonedVSMB := VSMBShare{
+		vm:       vm,
+		HostPath: vsmb.HostPath,
+		refCount: 1,
+		Name:     vsmb.Name,
+		Options:  vsmb.Options,
+		// TODO(ambarve) Do we need to include allowedFiles & guestPath here
```

Comment:
Yeah what happens for single file mounts? I would think that updating the compute system doc means that this directory will be reshared into the cloned VM right?